src/lread.c

```diff
@@ -4009,6 +4009,20 @@ check_obarray (Lisp_Object obarray)
   return obarray;
 }
 
+/* Like check_obarray, but for the global Vobarray. */
```

Comment:
So... at first I thought this would work. But if I replace

```rust
let obarray = unsafe { check_vobarray() };
```

with

```rust
let obarray = unsafe { check_obarray(globals.f_Vobarray) };
```

The bootstrapping process fails on the first call to `intern_1` in the C code, and Emacs will not build past the `temacs` stage, because `globals.f_Vobarray` is inexplicably `nil` at that point. This is a totally bizarre bug I haven't been able to track down. My best guess at the moment is that the `globals` object might have different contents for `obarray` during the bootstrap process and the regular Emacs incarnation.

A similar issue occurs with `globals.V_fpurify_flag` -- purify is not detected properly during bootstrapping, and symbols are not interned into pure storage when they should be (although the build does complete).

Any thoughts on this would be appreciated.